src/OpenTelemetry/Logs/LoggerProviderSdk.cs

```diff
@@ -16,14 +16,235 @@
+    private bool disposed;
+
     public LoggerProviderSdk(
         IServiceProvider serviceProvider,
         bool ownsServiceProvider)
     {
+        Debug.Assert(serviceProvider != null, "serviceProvider was null");
+
+        var state = serviceProvider!.GetRequiredService<LoggerProviderBuilderSdk>();
+        state.RegisterProvider(this);
+
+        this.ServiceProvider = serviceProvider!;
+
+        if (ownsServiceProvider)
+        {
+            this.ownedServiceProvider = serviceProvider as IDisposable;
+            Debug.Assert(this.ownedServiceProvider != null, "ownedServiceProvider was null");
+        }
+
+        OpenTelemetrySdkEventSource.Log.LoggerProviderSdkEvent("Building TracerProvider.");
+
+        var configureProviderBuilders = serviceProvider!.GetServices<IConfigureLoggerProviderBuilder>();
+        foreach (var configureProviderBuilder in configureProviderBuilders)
+        {
+            configureProviderBuilder.ConfigureBuilder(serviceProvider!, state);
+        }
+
+        StringBuilder processorsAdded = new StringBuilder();
+        StringBuilder instrumentationFactoriesAdded = new StringBuilder();
+
+        var resourceBuilder = state.ResourceBuilder ?? ResourceBuilder.CreateDefault();
+        resourceBuilder.ServiceProvider = serviceProvider;
+        this.Resource = resourceBuilder.Build();
+
+        foreach (var processor in state.Processors)
+        {
+            this.AddProcessor(processor);
+            processorsAdded.Append(processor.GetType());
+            processorsAdded.Append(';');
+        }
+
+        foreach (var instrumentation in state.Instrumentation)
+        {
+            this.instrumentations.Add(instrumentation.Instance);
+            instrumentationFactoriesAdded.Append(instrumentation.Name);
+            instrumentationFactoriesAdded.Append(';');
+        }
+
+        if (processorsAdded.Length != 0)
+        {
+            processorsAdded.Remove(processorsAdded.Length - 1, 1);
+            OpenTelemetrySdkEventSource.Log.LoggerProviderSdkEvent($"Processors added = \"{processorsAdded}\".");
+        }
+
+        if (instrumentationFactoriesAdded.Length != 0)
+        {
+            instrumentationFactoriesAdded.Remove(instrumentationFactoriesAdded.Length - 1, 1);
+            OpenTelemetrySdkEventSource.Log.LoggerProviderSdkEvent($"Instrumentations added = \"{instrumentationFactoriesAdded}\".");
+        }
+
+        OpenTelemetrySdkEventSource.Log.LoggerProviderSdkEvent("LoggerProviderSdk built successfully.");
+    }
+
+    public Resource Resource { get; }
+
+    public List<object> Instrumentations => this.instrumentations;
+
+    public BaseProcessor<LogRecord>? Processor { get; private set; }
+
+    public ILogRecordPool LogRecordPool => this.threadStaticPool ?? LogRecordSharedPool.Current;
+
+    public void AddProcessor(BaseProcessor<LogRecord> processor)
+    {
+        OpenTelemetrySdkEventSource.Log.LoggerProviderSdkEvent("Started adding processor.");
```

Comment:
We probably don't need this as this should cover it:

`OpenTelemetrySdkEventSource.Log.LoggerProviderSdkEvent($"Processors added = \"{processorsAdded}\".");`

Reply:
This is here because `AddProcessor` can be called after the provider has been created! But agree, it is wonky. I just took a stab at cleaning it all up. Pushed a couple `ToString` implementations into public API. LMK what you think about it now!